internal/service/alarms/internal/alarms_server.go

```diff
@@ -18,31 +18,98 @@ import (
 )
 
 type AlarmsServer struct {
+	// GlobalCloudID is the global O-Cloud identifier. Create subscription requests are blocked if the global O-Cloud identifier is not set
+	GlobalCloudID uuid.UUID
+	// AlarmsRepository is the repository for the alarms
 	AlarmsRepository *repo.AlarmsRepository
-	ResourceServer   *resourceserver.ResourceServer
+	// ResourceServer contains the resources server client and fetched resources
+	ResourceServer *resourceserver.ResourceServer
 }
 
 // AlarmsServer implements StrictServerInterface. This ensures that we've conformed to the `StrictServerInterface` with a compile-time check
 var _ api.StrictServerInterface = (*AlarmsServer)(nil)
 
-func (a *AlarmsServer) GetSubscriptions(ctx context.Context, request api.GetSubscriptionsRequestObject) (api.GetSubscriptionsResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+// GetSubscriptions handles an API request to fetch Alarm Subscriptions
+func (a *AlarmsServer) GetSubscriptions(ctx context.Context, _ api.GetSubscriptionsRequestObject) (api.GetSubscriptionsResponseObject, error) {
+	records, err := a.AlarmsRepository.GetAlarmSubscriptions(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Alarm Subscriptions due to issues with DB conn: %w", err)
+	}
+
+	objects := make([]api.AlarmSubscriptionInfo, len(records))
+	for _, record := range records {
+		objects = append(objects, models.ConvertSubscriptionModelToApi(record))
+	}
+
+	return api.GetSubscriptions200JSONResponse(objects), nil
 }
 
+// CreateSubscription handles an API request to create an Alarm Subscription
 func (a *AlarmsServer) CreateSubscription(ctx context.Context, request api.CreateSubscriptionRequestObject) (api.CreateSubscriptionResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+	// Block API if GlobalCloudID is not set
+	if a.GlobalCloudID == uuid.Nil {
+		return api.CreateSubscription409ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+			Detail: "provisioning of Alarm Subscriptions is blocked until the global O-Cloud identifier is set",
```

Comment:
We should probably generalize this to:

```
provisioning of Alarm Subscriptions is blocked until the SMO attributes are configured.
```

The SMO attributes would encompass not only the global cloud id but also the OAuth attributes which would also be required (eventually) for us to get a token and send a notification.

Reply:
Indeed, I will change it and add a TODO for Reachability check.

 O-RAN.WG6.ORCH-USE-CASES-R003-v11.00 suggest to perform a reachability check before creating a subscription. I saw that inventory has CABundleName but I couldn't find the token. I decided to leave it for later.

Reply:
ya, i still have to push those parameters down to the server.  Hopefully later this week or next.

Reply:
i like the reachability check idea